### PR TITLE
chore: Bb 317 update loading nmp files in bb

### PR DESCRIPTION
**Pull request by GDamaso**
# Description

BB now loads virtually any fert from a field with it's label onto the calculate nutrients list for the given field.

Please help me test and catch any major bug from this ticket. You can ask me for an .nmp file I created with all the fert options. I used it for mapping the ferts ids and testing.
 

## Type of changes

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have already been accepted and merged

## NMP Ferts

image

## BB loading NMP Ferts

image

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://nr-sustainment-capstone-2024-319-frontend.apps.silver.devops.gov.bc.ca)
- [Backend](https://nr-sustainment-capstone-2024-319-backend.apps.silver.devops.gov.bc.ca/api/health)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-sustainment-capstone-2024/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-sustainment-capstone-2024/actions/workflows/merge.yml)

### CODE REVIEW

**Review by GDamaso — COMMENTED** [review]
:ok_hand:

File: frontend/src/Utils/convertToNMP.ts [hunk 1]
```diff
@@ -35,7 +34,7 @@ const convertToNMP = (
   const newFields: NmpFieldInterface[] = newDetails.Fields.map((field: FieldDetailInterface) => {
     const newCrops: NmpCropInterface[] = field.Crops.map((crop: CropsDetailsInterface) => ({
       ...templateCropNMP,
-      id: crop.id + 1,
+      id: crop.id,
```
Comment (GDamaso) [inline frontend/src/Utils/convertToNMP.ts:37]: Tried matching our index to nmp... bad idea and it still loads fine, so better leave it as was before this addition, indexed at 0.
This conversion happens multiple times through the calculation, which compounded this addition.

**Review by GDamaso — COMMENTED** [review]
:ok_hand: 
Couldn't find any major bug, good enough for now

**Review by kcaparas1630 — COMMENTED** [review]
I'm not sure if because the openshift isn't recent to what you have running locally. But when I add fertilizers in the calculation. It doesn't show the Calculation List Component.

Screenshot_20240802_164137_Chrome

You'll see that the progress bar for calculation is green which means clicked. 

Also, I thought the return to calculation from export page is fixed. Since returning back, my changes weren't saved and I have to repeat the process again.

Here's a snippet. 
Screenshot_20240802_164207_Chrome


I can test this again locally on my laptop.

**Review by kcaparas1630 — COMMENTED** [review]
When I was loading the nmp file that you've sent me. I've noticed that it's starting on Crop ID 2
image
image

I'm not sure if it's the Openshift Link's not recently updated but here's what the fertilizer name looks in my end.
image

Again, this might be Openshift or a file problem. Creating a new calculation looks great.
Only thing I would like to mention is that, whenever you're returning back to calculation, fertilizer details are not being saved to the local storage, thus returning as empty object array and not the fertilizers that I've added.

**Comment by kcaparas1630** [comment]
tried reloading it and resetting my cache
here's what's happening on my end. it's returning the number value of that option.
image

**Comment by GDamaso** [comment]
Openshift is not updating the deployment, you gotta pull the branch and test it locally. Those bugs should have been fixed already:
image


The id's being subbed by 0 got fixed on a commit I just pushed.

Try it ou locally on desktop and let me know :smile:

**Comment by kcaparas1630** [comment]
Tested it out and it works well. One thing I would like to ask is that, since we have fertilizers printed on the list. Shouldn't it appear here in the Fertilizer CustomSelect Component as well? There's a lot of dry fertilizers but it doesn't let me choose another fertilizer to add to the field, or add that fertilizer in another field.
image

**Comment by GDamaso** [comment]
I totally agree with that new feature. Will create a new ticket for it. For now, we should move onto the blueberry calculation.

**Review by kcaparas1630 — APPROVED** [review]

**Pull request merged**